contractcourt/config.go

```diff
@@ -0,0 +1,113 @@
+package contractcourt
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcutil"
+)
+
+const (
+	// MinBudgetValue is the minimal budget that we allow when configuring
+	// the budget used in sweeping outputs. The actual budget can be lower
+	// if the user decides to NOT set this value.
+	//
+	// NOTE: This value is chosen so the liner fee function can increase at
+	// least 1 sat/kw per block.
+	MinBudgetValue btcutil.Amount = 1008
+
+	// MinBudgetRatio is the minimal ratio that we allow when configuring
+	// the budget ratio used in sweeping outputs.
+	MinBudgetRatio = 0.001
+
+	// DefaultBudgetRatio defines a default budget ratio to be used when
+	// sweeping inputs. This is a large value, which is fine as the final
+	// fee rate is capped at the max fee rate configured.
+	DefaultBudgetRatio = 0.5
+)
+
+// BudgetConfig is a struct that holds the configuration when offering outputs
+// to the sweeper.
+//
+//nolint:lll
+type BudgetConfig struct {
+	ToLocal      btcutil.Amount `long:"tolocal" description:"The amount in satoshis to allocate as the budget to pay fees when sweeping the to_local output. If set, the budget calculated using the ratio (if set) will be capped at this value."`
+	ToLocalRatio float64        `long:"tolocalratio" description:"The ratio of the value in to_local output to allocate as the budget to pay fees when sweeping it."`
+
+	AnchorCPFP      btcutil.Amount `long:"anchorcpfp" description:"The amount in satoshis to allocate as the budget to pay fees when CPFPing a force close tx using the anchor output. If set, the budget calculated using the ratio (if set) will be capped at this value."`
+	AnchorCPFPRatio float64        `long:"anchorcpfpratio" description:"The ratio of the value in time-sensitive HTLCs to allocate as the budget to pay fees when CPFPing a force close tx using the anchor output."`
```

Comment:
Yes! Updated the docs to make it more clear - think we still have room for improvement. One thing I've trying to figure out lately is, what values are time-sensitive here, and what a deadline means for CPFP anchor sweeps. It's currently defined as the `min(outgoing timeout, incoming timeout)`, while alternatively, it can be `min(outgoing's incoming timeout, incoming timeout)`.
- it makes sense to use an outgoing HTLC's incoming HTLC's refund timeout as the anchor deadline. When our outgoing HTLC has expired, it's ok to not sweep it as long as we are watching the mempool in case the remote goes on onchain with the preimage - in that case we'd extract it from the mempool and settle our incoming HTLC. We will only lose money when its corresponding incoming HTLC is expired and collected by the upstream. So all we care about is to make sure the outgoing HTLC is swept before that.
- It also makes sense to use an outgoing HTLC's refund timeout because the force close tx must be confirmed before we can start sweeping the outgoing HTLC. The anchor's responsibility is to make sure FC confirms in time. This is the current design, it makes things simpler.

The other question is what is the value under protection? If we have two HTLCs that have very different timeouts, one is 40 and the other is 1000, then it doesn't make sense to give them the same weight when deciding the time-sensitive value. Something to optimize in future PRs.

To zoom out a bit, the whole design is really just the following equation,
- `totalFee = closing_tx_fee + to_local_fee + 1st_level_htlc_fee + 2nd_level_htlc_fee`

And we want to cap the `totalFee` at the total budgets, and, if using default params, should never exceed the balance that can be taken back of a given channel. There's also a relationship of budget ratios that are not enforced in this PR (again future PRs), but the general idea is there exists a tradeoff when allocating budgets - if you put more budgets in CPFP, you'd have fewer budgets in HTLCs, since the channel balance is the limit.